fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CancelCompactionStmtAnalyzer.java

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.analyzer;
+
+import com.starrocks.analysis.BinaryPredicate;
+import com.starrocks.analysis.BinaryType;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.IntLiteral;
+import com.starrocks.analysis.SlotRef;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.sql.ast.AstVisitor;
+import com.starrocks.sql.ast.CancelCompactionStmt;
+
+public class CancelCompactionStmtAnalyzer {
+
+    public static void analyze(CancelCompactionStmt statement, ConnectContext context) {
+        new CancelCompactionStmtAnalyzerVisitor().analyze(statement, context);
+    }
+
+    static class CancelCompactionStmtAnalyzerVisitor extends AstVisitor<Void, ConnectContext> {
+
+        private long txnId;
+
+        public CancelCompactionStmtAnalyzerVisitor() {
+            this.txnId = 0;
+        }
+
+        public void analyze(CancelCompactionStmt statement, ConnectContext context) {
+            visit(statement, context);
+        }
+
+        @Override
+        public Void visitCancelCompactionStatement(CancelCompactionStmt statement, ConnectContext context) {
+            analyzeWhereClause(statement, context);
+            return null;
+        }
+
+        private void analyzeWhereClause(CancelCompactionStmt statement, ConnectContext context) {
+            Expr whereClause = statement.getWhereClause();
+            boolean valid = true;
+            do {
+                if (whereClause == null) {
+                    valid = false;
+                    break;
+                }
+
+                if (whereClause instanceof BinaryPredicate) {
+                    BinaryPredicate binaryPredicate = (BinaryPredicate) whereClause;
+                    if (binaryPredicate.getOp() != BinaryType.EQ) {
+                        valid = false;
+                        break;
+                    }
+                } else {
+                    valid = false;
+                    break;
+                }
+
+                // left child
+                if (!(whereClause.getChild(0) instanceof SlotRef)) {
+                    valid = false;
+                    break;
+                }
+                if (!((SlotRef) whereClause.getChild(0)).getColumnName().equalsIgnoreCase("txn_id")) {
```

Comment:
How does the user to get know what the `txn_id` is?

Reply:
show proc '/compactions' will show all running compactions infomation :)